src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -495,4 +500,135 @@ public void close() throws Exception {
+            inspectionPortEntity.setEgressPort((NetworkElementEntity) inspectionPort.getEgressPort());
+            inspectionPortEntity.setIngressPort((NetworkElementEntity) inspectionPort.getIngressPort());
+
+            return this.em.merge(inspectionPortEntity);
+        });
+    }
+
+    public List<String> getInspectionPortsIds() throws Exception {
+
+        return this.txControl.supports(() -> {
+
+            List<? extends InspectionPortEntity> elements = this.utils.txInspectionPortEntities();
+            List<String> elementList = new ArrayList<String>();
+
+            if (!elements.isEmpty()) {
+                elementList = elements.stream().map(InspectionPortEntity::getElementId).collect(Collectors.toList());
+            }
+
+            return elementList;
+        });
+    }
+
+    public NetworkElement registerPort(NetworkElement nwElement) throws Exception {
+
+        NetworkElementEntity element = this.utils.txNetworkElementEntityByElementId(nwElement.getElementId());
+        if (element != null) {
+            String msg = String.format("Network element already exists id: %s\n", nwElement.getElementId());
+            LOG.error(msg);
+            throw new IllegalArgumentException(msg);
+        }
+
+        return this.txControl.required(() -> {
+            NetworkElementEntity portElement = new NetworkElementEntity();
+            portElement.setElementId(nwElement.getElementId());
+            portElement.setMacAddresses(nwElement.getMacAddresses());
+            portElement.setPortIPs(nwElement.getPortIPs());
+            portElement.setDeviceOwnerId(((NetworkElementEntity) nwElement).getDeviceOwnerId());
+            portElement.setParentId(nwElement.getParentId());
+
+            this.em.merge(portElement);
+
+            return portElement;
+        });
+    }
+
+    public NetworkElement updatePort(NetworkElement nwElement) throws Exception {
+
+        NetworkElementEntity element = this.utils.txNetworkElementEntityByElementId(nwElement.getElementId());
+        if (element == null) {
+            String msg = String.format("Cannot find the network element id: %s\n", nwElement.getElementId());
+            LOG.error(msg);
+            throw new IllegalArgumentException(msg);
+        }
+
+        return this.txControl.required(() -> {
+            element.setMacAddresses(nwElement.getMacAddresses());
+            element.setPortIPs(nwElement.getPortIPs());
+            element.setDeviceOwnerId(((NetworkElementEntity) nwElement).getDeviceOwnerId());
+            element.setParentId(nwElement.getParentId());
+
+            this.em.merge(element);
+
+            return element;
+        });
+    }
+
+    public void deletePort(String id) throws Exception {
+
+        this.txControl.required(() -> {
+            NetworkElementEntity element = this.utils.txNetworkElementEntityByElementId(id);
+
+            if (element == null) {
+                LOG.warn(String.format(
```

Comment:
with slf4j logging no longer needs to use string.format
Use below if we are just logging the message and not wrapping it with an exception

`LOG.warn("Attempt to delete network element for id {} and network element not found, no-op.", id);`